### ISSUE TITLE
TorchVision Roadmap - 2021 H1

### ISSUE THREAD

**Issue by datumbox** [pytorch/vision#3221]
This issue is be used to keep track of the roadmap of TorchVision and get feedback from our users.

For the 2021-H2 roadmap, please see https://github.com/pytorch/vision/issues/4187

- Streamlined data pipelines:
	- [x] Improved io.image with new supported types and conversions (#2984, #2988, #3024, #3069)
	- [x] Add Auto-augment input transformation (#3123)
	- [x] Add preset preprocessing transforms on pretrained models (#3317)
	- [x] Add GPU/CUDA ops for image decoding (jpeg) (https://github.com/pytorch/vision/pull/3792)
	- [x] Add and optimize native uint8 support for interpolate in PyTorch
- Mobile support:
	- [x] Refactor C++ codebase & make it mobile-friendly (#2885, #2891, #2892, #2893, #2944, #2945, #3011, #3097, #3135, #3135, #3146, #3156, #3154, #3163, #3218, #3311, #3312) 
	- [x] Add mobile-ready models for detection (#3253, #3265, #3403, #3757, #3760)
	- [x] Add mobile-ready models for segmentation (#3276)
	- [x] Support D2Go OSS
	- [x] Add quint8 implementations for nms and RoIAlign (https://github.com/pytorch/vision/pull/3601, https://github.com/pytorch/vision/pull/3624)
	- [ ] Rotated Boxes (#2761) (moved to 2022H2 #4187)
- Add newer models for classification:
	- [x] MobileNetV3 (#3252, #3354)
	- [x] MobileNetV3 Quantized (#3323)

**Comment by masahi** [comment]
Can we revisit various workarounds for tracing Faster RCNN / MaskRCNN ? There are many such special code-path for tracing mode, to support ONNX export. These code were added one year ago when PyTorch ONNX had many limitation. Especially the one below does repeated concat in a loop so it is not efficient. 

https://github.com/pytorch/vision/blob/3d60f498e71ba63b428edb184c9ac38fa3737fa6/torchvision/models/detection/roi_heads.py#L454-L461

Now that PyTorch ONNX export has support for lists and also inplace assignment, I think many of those workarounds can be removed. cc @fmassa @lara-hdr

**Comment by fmassa** [comment]
@masahi I would be happy to accept a PR to improve the ONNX export path, and removing as many workarounds as possible would be awesome. cc @neginraoof for thoughts.

**Comment by oke-aditya** [comment]
Just linking a few trackers here #2707 , #2980

**Comment by vfdev-5** [comment]
How about adding [RandAugment](https://arxiv.org/abs/1909.13719v2), in addition to already implemented AutoAugment ?

**Comment by nSircombe** [comment]
Hi,

I was just wondering if there are any plans to add support for the QNNpack backend to the quantised ResNet50 model? (this would enable quantised ResNet50 for AArch64, which isn't supported by the FBGEMM backend).

**Comment by zhiqwang** [comment]
> - Mobile support:
>   - [x] Support D2Go OSS

Is the newly released [d2go](https://github.com/facebookresearch/d2go) mentioned here?

**Comment by oke-aditya** [comment]
Thoughts about [FastAutoAugment ](https://github.com/kakaobrain/fast-autoaugment) maybe useful for faster transforms

**Comment by jamt9000** [comment]
> * [ ]  Add GPU/CUDA ops for image decoding (jpeg)

Implemented in https://github.com/pytorch/vision/pull/3792

**Comment by oke-aditya** [comment]
Rotated Boxes RFC #2761.

**Comment by fmassa** [comment]
@nSircombe 

> I was just wondering if there are any plans to add support for the QNNpack backend to the quantised ResNet50 model? (this would enable quantised ResNet50 for AArch64, which isn't supported by the FBGEMM backend).

can you please open a new issue so that we can track this?

@zhiqwang 

> Is the newly released d2go mentioned here?

yes, exactly

@oke-aditya 

> Thoughts about FastAutoAugment ?

Can you open a new issue so that we can track it down and discuss it there?

**Comment by nSircombe** [comment]
Hi @fmassa,

I currently have https://github.com/pytorch/vision/issues/3362 open (I should have added this in my original comment, sorry), will that suffice?

**Comment by Atze00** [comment]
Is there any interest in adding MoViNets or some other state of the art models in video understanding?
Are models valuable also if they are not pretrained/someone has the possibility to train them before pushing to Pytorch?
Otherwise these models may use imported weights from tf, with the overhead of same padding, but I'm not convinced about this option.

**Comment by NicolasHug** [comment]
I updated the post with our roadmap for H2. Comments and suggestions are welcome :) !

**Comment by NicolasHug** [comment]
@Atze00 there's no current plan to support MoViNets but regarding video understanding, we're planning to support Optical Flow with a RAFT implementation

>  Are models valuable also if they are not pretrained/someone has the possibility to train them before pushing to Pytorch?

Could you clarify what you mean? I'm not sure this answers your question but in general, when we implement a model in torchvision we provide the architecture, the pre-trained weights, and also some training receipe

**Comment by Atze00** [comment]
@NicolasHug My question was based on the fact that I've been working on implementing [movinets in pytorch](https://github.com/Atze00/MoViNet-pytorch), using the weights released by the authors in TF. I've been wondering if the code alone would be enough to discuss about a possible addition of this architecture into the roadmap. Unfortunately I don't have the resources or the experience necessary to reproduce state of the art results in video understanding.

**Comment by datumbox** [comment]
@fmassa @NicolasHug Shall we close this issue in favor of #4187? 

Since only the rotated boxes RFC is still open we could consider moving it on H2 or omit it if nobody wants to pick it up.

**Comment by datumbox** [comment]
Moved the pending to H2 at #4187, closing this issue.
